src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,48 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public abstract class Contact {
+	
+	private final String value;
+	private final boolean isPrivate;
+	
+    /**
+     * Validates given contact using the given regular expression.
+     *
+     * @throws IllegalValueException if given email address string is invalid.
+     */
+	public Contact(String value, boolean isPrivate, String regex, String message_constraints) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        String trimmedValue = value.trim();
+        if (!isValidContact(trimmedValue, regex)) {
+            throw new IllegalValueException(message_constraints);
+        }
+        this.value = trimmedValue;
+	}
+	
+    /**
+     * Checks if a given string is a valid contact.
+     */
+	public boolean isValidContact(String test, String regex) {
```

Comment:
Is there anyway to avoid passing in the regex again? Also, as this is a public method, user of the class / subclasses can really mess thing up if they pass in different regex strings...